interop-twitter/jvm/src/test/scala/scalaz/zio/interop/TwitterSpec.scala

```diff
@@ -47,4 +55,26 @@ class TwitterSpec(implicit ee: ExecutionEnv) extends TestRuntime {
     unsafeRun(task <* ZIO.unit.delay(3.seconds)) must beNone
     value.get() ==== 0
   }
+
+  private def evaluateToSuccessfulFuture = {
+    Await.result(this.unsafeRunToTwitterFuture(Task.succeed(1))) ==== 1
+    Await.result(this.unsafeRunToTwitterFuture(Task.succeed(2).delay(100.millis))) ==== 2
+  }
+
+  private def evaluateToFailedFuture = {
+    val e    = new Exception
+    val task = ZIO.unit.delay(100.millis) *> Task.fail(e).unit
+
+    Await.result(this.unsafeRunToTwitterFuture(task)) must throwAn(e)
+  }
+
+  private def evaluateToInterruptedFuture = {
+    val value         = new AtomicInteger(0)
+    val futureTimeout = TwitterDuration.fromMilliseconds(100)
+
+    val task = ZIO.unit.delay(500.millis) *> Task.effect(value.incrementAndGet())
```

Comment:
These sleeps will make tests non-deterministic. We should use latches to eliminate the need for sleeping. Let me known if you need a hand with that.

Reply:
I'll have a go

Reply:
One of the test is still flaky, need some advices